agent/consul/internal_endpoint.go

```diff
@@ -323,12 +323,26 @@ func (m *Internal) GatewayServices(args *structs.ServiceSpecificRequest, reply *
 			var index uint64
 			var services structs.GatewayServices
 
-			switch args.ServiceKind {
-			case structs.ServiceKindTerminatingGateway:
-				index, services, err = state.TerminatingGatewayServices(ws, args.ServiceName, &args.EnterpriseMeta)
+			supportedGateways := []string{structs.IngressGateway, structs.TerminatingGateway}
```

Comment:
Maybe move this to a package variable?

Reply:
What's your reasoning behind that? I kind of like having this declared close to the loop we use it in.

Reply:
Mostly to avoid creating a slice every time the function is called when it is known to be a fixed immutable value. This is probably my Java6-era optimization leaking through, so you can probably also freely disregard.